lib/http/HttpServer.ts

```diff
@@ -54,35 +54,43 @@ class HttpServer {
       return;
     }
 
-    let resJson: any;
-    try {
-      switch (req.url) {
-        case '/resolveraiden':
-          resJson = await this.httpService.resolveHashRaiden(reqJson);
-          break;
-        default:
-          res.writeHead(404);
-          res.end();
-          break;
-      }
-      res.writeHead(200, { 'Content-Type': 'application/json' });
-      res.end(JSON.stringify(resJson));
-    } catch (err) {
-      switch (err.code) {
-        case swapErrorCodes.INVALID_RESOLVE_REQUEST:
-        case errorCodes.INVALID_ARGUMENT:
-          res.writeHead(400);
-          res.end(err.message);
-          break;
-        case swapErrorCodes.PAYMENT_HASH_NOT_FOUND:
-          res.writeHead(404);
-          res.end(err.message);
-          break;
-        default:
-          res.writeHead(500);
-          res.end(JSON.stringify(err));
-          break;
-      }
+    switch (req.url) {
+      case '/resolveraiden':
+        try {
+          const resJson = await this.httpService.resolveHashRaiden(reqJson);
+          res.writeHead(200, { 'Content-Type': 'application/json' });
+          res.end(JSON.stringify(resJson));
+        } catch (err) {
+          switch (err.code) {
+            case swapErrorCodes.INVALID_RESOLVE_REQUEST:
+            case errorCodes.INVALID_ARGUMENT:
+              res.writeHead(400);
+              res.end({ message: err.message, retry: false });
+              break;
+            case swapErrorCodes.PAYMENT_HASH_NOT_FOUND:
+              res.writeHead(404);
+              res.end({ message: err.message, retry: false });
+              break;
+            case swapErrorCodes.FINAL_PAYMENT_ERROR:
+            case swapErrorCodes.PAYMENT_REJECTED:
+              res.writeHead(500);
+              res.end({ message: err.message, retry: false });
+              break;
+            case swapErrorCodes.UNKNOWN_PAYMENT_ERROR:
+              res.writeHead(503);
+              res.end({ message: err.message, retry: true });
+              break;
+            default:
+              res.writeHead(503);
+              res.end({ err, retry: true });
+              break;
+          }
+        }
+        break;
+      default:
+        res.writeHead(404);
```

Comment:
Isn't is safer to return 400 here?

Reply:
I think 404 is the standard HTTP status code when a request targets a path that does not exist or is not recognized. But part of what I'm going for here is that the requester (raiden) should not rely on the status code to determine whether a resolve request failure is permanent or not.